File: frontend/services/statistics/types/statistics.ts
/** Mirrors backend/app/schemas/statistics.py exactly. */
export type StatisticsTotals = {
  guests: number;
  scheduled_interviews: number;
  completed_interviews: number;
  questions: number;
  content_drafts: number;
  approved_content: number;
  designs: number;
  design_slides: number;
};

export type PlatformCount = {
  platform: "linkedin" | "x" | "instagram" | "general";
  count: number;
};

export type ActivityPoint = {
  /** "YYYY-MM-DD" */
  date: string;
  count: number;
};

export type StatisticsOverview = {
  totals: StatisticsTotals;
  content_by_platform: PlatformCount[];
  activity: ActivityPoint[];
};

File: backend/app/schemas/statistics.py
from pydantic import BaseModel


class StatisticsTotals(BaseModel):
    guests: int
    scheduled_interviews: int
    completed_interviews: int
    questions: int
    content_drafts: int
    approved_content: int
    designs: int
    design_slides: int


class PlatformCount(BaseModel):
    platform: str
    count: int


class ActivityPoint(BaseModel):
    date: str
    count: int


class StatisticsOverviewResponse(BaseModel):
    totals: StatisticsTotals
    content_by_platform: list[PlatformCount]
    activity: list[ActivityPoint]
